include/envoy/router/scopes.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/config/config_provider.h"
+#include "envoy/router/router.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * The scoped routing configuration.
+ */
+class ScopedConfig : public Envoy::Config::ConfigProvider::Config {
+public:
+  ~ScopedConfig() override = default;
+
+  /**
+   * Based on the incoming HTTP request headers, returns the configuration to use for selecting a
+   * target route.
+   * @param headers the request headers to match the scoped routing configuration against.
+   * @return ConfigConstSharedPtr the router's Config matching the request headers.
+   */
+  virtual ConfigConstSharedPtr getRouterConfig(const Http::HeaderMap& headers) const PURE;
```

Comment:
This returns an `Envoy::Router::Config`, hence the name. However, I agree it is fairly confusing since the documentation refers to it by yet another name ([route table](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http_routing#route-table)).

I am fine switching to `getRouteTable` or even `getRouteConfiguration` if you think either of those more directly convey the meaning.

Reply:
`getRouteConfiguration` or even the shorter `getRouteConfig` would be clearer to me.

Reply:
Done.